avm/ptn/app-service-lza/hosting-environment/modules/app-service/app-service.module.bicep

```diff
@@ -0,0 +1,332 @@
+// Dependencies //
+// ============ //
+
+// var siteConfigConfigurationMap = {
+//   windowsNet6: {
+//     metadata: [
+//       {
+//         name: 'CURRENT_STACK'
+//         value: 'dotnet'
+//       }
+//     ]
+//     netFrameworkVersion: 'v6.0'
+//     use32BitWorkerProcess: false
+//   }
+//   windowsNet7: {
+//     metadata: [
+//       {
+//         name: 'CURRENT_STACK'
+//         value: 'dotnet'
+//       }
+//     ]
+//     netFrameworkVersion: 'v7.0'
+//     use32BitWorkerProcess: false
+//   }
+//   windowsAspNet486: {
+//     metadata: [
+//       {
+//         name: 'CURRENT_STACK'
+//         value: 'dotnet'
+//       }
+//     ]
+//     netFrameworkVersion: 'v4.0'
+//     use32BitWorkerProcess: false
+//   }
+//   linuxJava17Se: {
+//     linuxFxVersion: 'java|17-java17'
+//     use32BitWorkerProcess: false
+//   }
+//   linuxNet8: {
+//     linuxFxVersion: 'dotnetcore|8.0'
+//     use32BitWorkerProcess: false
+//   }
+//   linuxNet7: {
+//     linuxFxVersion: 'dotnetcore|7.0'
+//     use32BitWorkerProcess: false
+//   }
+//   linuxNet6: {
+//     linuxFxVersion: 'dotnetcore|6.0'
+//     use32BitWorkerProcess: false
+//   }
+//   linuxNode18: {
+//     linuxFxVersion: 'node|18-lts'
+//     use32BitWorkerProcess: false
+//   }
+// }
+
+module ase './ase.module.bicep' = if (deployAseV3) {
+  name: '${uniqueString(deployment().name, location)}-ase'
+  params: {
+    name: aseName
+    location: location
+    tags: tags
+    subnetResourceId: subnetIdForVnetInjection
+    zoneRedundant: zoneRedundant
+    allowNewPrivateEndpointConnections: true
+    virtualNetworkLinks: virtualNetworkLinks
+    diagnosticSettings: aseDiagnosticSettings
+  }
+}
+
+module appInsights 'br/public:avm/res/insights/component:0.4.1' = {
+  name: '${uniqueString(deployment().name, location)}-appInsights'
+  params: {
+    name: 'appi-${webAppName}'
+    location: location
+    tags: tags
+    workspaceResourceId: logAnalyticsWorkspaceResourceId
+    applicationType: 'web'
+    publicNetworkAccessForIngestion: 'Enabled'
+    publicNetworkAccessForQuery: 'Enabled'
+    retentionInDays: 90
+    samplingPercentage: 100
+  }
+}
+
+module plan 'br/public:avm/res/web/serverfarm:0.2.4' = {
+  name: '${uniqueString(deployment().name, location, 'webapp')}-plan'
+  params: {
+    name: appServicePlanName
+    location: location
+    tags: tags
+    skuName: sku
+    zoneRedundant: zoneRedundant
+    kind: (webAppBaseOs =~ 'linux') ? 'Linux' : 'Windows'
+    perSiteScaling: perSiteScaling
+    maximumElasticWorkerCount: (maximumElasticWorkerCount < 3 && zoneRedundant) ? 3 : maximumElasticWorkerCount
+    reserved: serverOS == 'Linux'
+    targetWorkerCount: (targetWorkerCount < 3 && zoneRedundant) ? 3 : targetWorkerCount
+    targetWorkerSize: targetWorkerSize
+    appServiceEnvironmentId: deployAseV3 ? ase.outputs.resourceId : ''
+    diagnosticSettings: servicePlanDiagnosticSettings
+  }
+}
+
+module webApp 'br/public:avm/res/web/site:0.9.0' = {
```

Comment:
Please ensure (this should also fail the latest static tests) to always pass the `enableTelemetry` parameter to all referenced AVM modules.

Reason being: It **must** always be possible to easily disable telemetry for modules. The problem by not passing the `enableTelemetry` value through is that its default in the module is `true`. So even if a consumer wants to deploy this module with telemetry off - it would currently only disable the telemetry deployment for the `hosting environment` module, but not e.g., the `web/site` module (and others) you're using here.

Fixing this is as easy as passing the parameter from the top-level template down to the `/modules/` you defined and also here just pass the value to the AVM modules like  `enableTelemetry: enableTelemetry`> 🚀

Reply:
Same comment applies to other files

Reply:
addressed